### PR TITLE
Patch/rework for app

### PR DESCRIPTION
**Pull request by aaron-christian-nttd**
This PR is a refactor of the `launch service` group of commands. It reworks the commands to work with the new template from this PR: https://github.com/launchbynttdata/lcaf-template-terragrunt/pull/3

## General 
- More reorg and moving between constants/ and config/ items.
- Moving functions into more defined dirs than just `common`
- Adding dry-run to all 3 commands. when adding `--dry-run` flag, it will go through the operations and report on what it would do.
- Adding a `override_default` function. This function looks for any set ENV vars and returns that. If not, will look for a local `.launch_config` with its key in {override:{<var_name>: <var_data>}}
- commenting out `az` specific functions. This will need to be rework later.
- Adding testing stubs for the new functions. Will be completed in later PR after demo. 

## `launch service create`
- refactoring to work with this PR: https://github.com/launchbynttdata/lcaf-template-terragrunt/pull/3 THis is the new template format to simplify this process.

## `launch service update`
- When updating a service, the command was simplified to look for a local `.launch_config` file if no `--in-file` was supplied.

## `launch service generate`
- when generating now, it will utilizing a ./launch/build` folder and use this folder for build activities. Will no longer make a `singleRun` dir.
- runs `launch service clean` before running to ensure a clean build folder. 

## `launch service clean`
- changing from `cleanup` to `clean`
- having it look for and remove `./launch/build`

### CODE REVIEW

File: src/launch/lib/service/template/functions.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    template_path: Path,
+    destination_dir: str,
+    file_name: str,
+    template_data: dict = {"data": None},
+    dry_run: bool = True,
+) -> None:
+    if not template_data.get("data"):
+        template_data["data"] = {}
+
+    env = Environment(loader=FileSystemLoader(template_path.parent))
```
[the remark this answers is not in the document]
Comment (chris11-taylor-nttd) [reply src/launch/lib/service/template/functions.py:161]: We're probably not vulnerable to XSS given this isn't a webapp, but setting autoescape is a good idea regardless.

File: src/launch/lib/service/template/functions.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        str(destination_dir)[(str(destination_dir).find(PLATFORM_SRC_DIR_PATH) + 9) :],
+    )
+    output = template.render(template_data)
+    destination_path = destination_dir / file_name
+
+    if output.strip():
+        with open(destination_path, "w") as f:
+            if dry_run:
+                click.secho(
+                    f"[DRYRUN] Rending template, would have saved rendered file: {destination_path}",
```
Comment (chris11-taylor-nttd) [inline src/launch/lib/service/template/functions.py:175]:
```suggestion
                    f"[DRYRUN] Rendering template, would have saved rendered file: {destination_path}",
```

There are a few other typos in this vein.

File: test/unit/cli/service/clean/test_clean.py
Comment (chris11-taylor-nttd) [inline test/unit/cli/service/clean/test_clean.py:1]: This and the other test stubs need to be implemented.

**Review by chris11-taylor-nttd — COMMENTED** [review]
Aside from some tests that need implementations and a few minor tweaks elsewhere, this looks really good. I'm happy to provide my approval once the tests are completed.

One consideration for the future that we might address in a later revision: our .launch_config file doesn't really get validated prior to those values being plugged in elsewhere, which means that an invalid config file could result in nasty stack traces raised from distantly-related parts of the software, making it difficult for a user to understand what has gone wrong. 

We could write our own validation logic but I think a much nicer solution would be to use a third-party library like `pydantic` for validation purposes, and then we can inject the resultant validated object into any dependent code. Having an indication that a config is invalid before we try to use it can keep us from getting into a potentially invalid state.

File: src/launch/lib/service/template/functions.py [hunk 3]
```diff
@@ -158,7 +158,7 @@ def render_jinja_template(
     if not template_data.get("data"):
         template_data["data"] = {}
 
-    env = Environment(loader=FileSystemLoader(template_path.parent))
+    env = Environment(loader=FileSystemLoader(template_path.parent), autoescape=False)
```
Comment (chris11-taylor-nttd) [inline src/launch/lib/service/template/functions.py:161]: This is what the CodeQL report is asking for:

```suggestion
    env = Environment(loader=FileSystemLoader(template_path.parent), autoescape=True)
```

**Review by chris11-taylor-nttd — APPROVED** [review]

**Review by bryce-lynn-nttd — APPROVED** [review]

**Review by aarti-joshi-nttd — APPROVED** [review]

**Pull request merged**